xdem/coreg/base.py

```diff
@@ -1259,57 +1597,233 @@ def is_affine(self) -> bool:
 
         return self._is_affine
 
+    @property
+    def is_translation(self) -> bool | None:
+
+        # If matrix exists in keys, or can be derived from to_matrix(), we conclude
+        if "matrix" in self._meta["outputs"]["affine"].keys():
+            matrix = self._meta["outputs"]["affine"]["matrix"]
+        else:
+            try:
+                matrix = self.to_matrix()
+            # Otherwise we can't yet and return None
+            except (AttributeError, ValueError, NotImplementedError):
+                self._is_translation = None
+                return None
+
+        # If the 3x3 rotation sub-matrix is the identity matrix, we have a translation
+        return np.allclose(matrix[:3, :3], np.diag(np.ones(3)), rtol=10e-3)
+
     @property
     def meta(self) -> CoregDict:
         """Metadata dictionary of the coregistration."""
 
         return self._meta
 
+    @overload
+    def info(self, verbose: Literal[True] = ...) -> None:
+        ...
+
+    @overload
+    def info(self, verbose: Literal[False]) -> str:
+        ...
+
+    def info(self, verbose: bool = True) -> None | str:
+        """Summarize information about this coregistration."""
+
+        # Map each key name to a descriptor string
+        dict_key_to_str = {
```

Comment:
Should we not have a separate list outside of this class? If one adds a new argument to an existing coreg method or a new coreg method, one has to add the argument here and also in the corresponding dictionary no? I don't know if there is a simple way to combine both...

Reply:
Agreed! Did this, and added a test in `test_base` that will raise an error if a key in the `info()` dictionary has not been added :smile: